### PR TITLE
Improve readme

### PR DESCRIPTION
**Pull request by StevenXL**
A few updates on the commit. No radical changes - just added some links, a small "gif" to get folks excited about the product, and moved some sections around. (For example, moved the dependencies section further down the page to de-emphasize them).

No pride of authorship, so please change these as you see fit. Great program - I'm looking forward to using it many times as I learn Python.

### CODE REVIEW

**Comment by sebastinas** [comment]
Please squash the commits together.

File: README.rst [hunk 1]
```diff
@@ -2,166 +2,159 @@
...
+* babel (optional, for internationalization)
+* watchdog (optional, for monitoring imported modules for changes)
+* jedi (optional, for experimental multiline completion)
+
+If you are using Python 2 before 2.7.7, the following dependency is also
+required:
+
+* `requests[security]`_ 
+
+If you have problems installing cffi which is needed by pyOpenSSL, please take a look at `cffi docs`_.
```
Comment (sebastinas) [inline README.rst]: I think we should refer to the homepage here instead. It's easier to document all installation issues there than keep this section up-to-date.

File: README.rst [hunk 2]
```diff
@@ -170,3 +163,17 @@ Documentation
...
+.. _homepage: http://www.bpython-interpreter.org
+.. _full documentation: http://docs.bpython-interpreter.org/
+.. _cffi docs: https://cffi.readthedocs.org/en/release-0.8/#macos-x
+.. _issues tracker: http://github.com/bpython/bpython/issues/
+.. _pip: https://pip.pypa.io/en/latest/index.html
+.. _project homepage: http://bpython-interpreter.org/community
+.. _community: http://docs.bpython-interpreter.org/community.html
+.. _Robert Anthony Farrell: robertanthonyfarrell@gmail.com
+.. _bpython: http://www.bpython-interpreter.org/
+.. _requests[security]: https://pypi.python.org/pypi/requests/
```
Comment (sebastinas) [inline README.rst]: Why is this a link and the other dependencies are not?

**Comment by sebastinas** [comment]
Please wrap lines at 80 characeters.

**Comment by StevenXL** [comment]
I've squashed the commits, wrapped lines at 80 characters, and removed a few links to make the items that have link and don't have link more consistent.

File: README.rst [hunk 3]
```diff
@@ -2,12 +2,87 @@
...
+    :backlinks: none
+
+==========================
+Installation & Basic Usage
+==========================
+If you have `pip`_ installed, you can simply run:
+
+.. code-block:: bash
+
+    $ sudo pip install bpython
```
Comment (sebastinas) [inline README.rst]: Please do not recommend this. Never.

File: README.rst [hunk 4]
```diff
@@ -33,140 +106,69 @@ bpython-urwid
...
 
+============
 Curses Notes
-------------
-The curses used has a bug where the colours are displayed incorrectly:
- * red  is swapped with blue
- * cyan is swapped with yellow
+============
+The curses used has a bug where the colours are displayed incorrectly: * red  is
+swapped with blue * cyan is swapped with yellow
```
Comment (sebastinas) [inline README.rst]: Please keep the list here.

File: README.rst (hunk 3 above)
Comment (StevenXL) [inline README.rst]: @sebastinas can you be a bit more explicit about what I should not recommend? Is is the fact that we are using the `sudo` command? I have no problem changing this, but I don't understand the issue, so I don't know what to change it to.

File: README.rst (hunk 4 above)
Comment (StevenXL) [inline README.rst]: Will do - this was a typo from reformatting the doc to wrap at 80 chars. I thought I caught them all. I'll change this tonight (unfortunately, real work beckons).

File: README.rst [hunk 5]
```diff
@@ -33,140 +106,69 @@ bpython-urwid
...
-    http://docs.bpython-interpreter.org/
 
-Developer documentation
-    http://docs.bpython-interpreter.org/contributing.html
+.. _homepage: http://www.bpython-interpreter.org
+.. _full documentation: http://docs.bpython-interpreter.org/
+.. _cffi docs: https://cffi.readthedocs.org/en/release-0.8/#macos-x
+.. _issues tracker: http://github.com/bpython/bpython/issues/
+.. _pip: https://pip.pypa.io/en/latest/index.html
+.. _project homepage: http://bpython-interpreter.org/community
```
Comment (sebastinas) [inline README.rst]: Remove the community part.

File: README.rst [hunk 6]
```diff
@@ -33,140 +106,69 @@ bpython-urwid
...
+For known bugs please see bpython's `issues tracker`_ at github.
 
-For any other ways of communicating with bpython users and devs you can find us
-at the community page on the projects homepage:
-http://bpython-interpreter.org/community
+======================
+Contact & Contributing
+======================
+I hope you find it useful and please feel free to submit any bugs/patches
+suggestions to `Robert Athony Farrell`_ or place them on the github
```
Comment (sebastinas) [inline README.rst]: @bobf: Do you still want to receive bugs/patches per mail or should we just refer to the bug tracker and the mailing list?

File: README.rst (hunk 3 above)
Comment (sebastinas) [inline README.rst]: `sudo` is problematic here. See http://notes.pault.ag/debian-python/ for some background. But in short: `sudo pip install` will break horribly if you also install Python modules with a package manager.

File: README.rst (hunk 5 above)
Comment (StevenXL) [inline README.rst]: done

File: README.rst (hunk 4 above)
Comment (StevenXL) [inline README.rst]: done

File: README.rst [hunk 7]
```diff
@@ -33,140 +101,69 @@ bpython-urwid
...
-
-I hope you find it useful and please feel free to submit any bugs/patches (yeah
-right)/suggestions to:
-robertanthonyfarrell@gmail.com
-or place them at the github issue page for this project at:
-http://github.com/bpython/bpython/issues/
+==========
+Known Bugs
+==========
+For known bugs please see bpython's `issues tracker`_ at github.
```
Comment (sebastinas) [inline README.rst]: It's GitHub. Please also refer to http://bpython-interpreter.org/issues.html (should go live later today).

File: README.rst [hunk 8]
```diff
@@ -18,14 +93,7 @@ Dependencies
...
 
-If you are using Python 2 before 2.7.7, the following dependency is also
-required:
-
-* requests[security]
-
-If you have problems installing cffi which is needed by pyOpenSSL,
-please take a look at https://cffi.readthedocs.org/en/release-0.8/#macos-x.
-
+Please refer to our documentation for further details visite our `homepage`_.
```
Comment (sebastinas) [inline README.rst]: There is something missing here.

File: README.rst (hunk 7 above)
Comment (sebastinas) [inline README.rst]: That should be http://bpython-interpreter.org/known-issues-and-faq.html.

**Comment by sebastinas** [comment]
Once my two remaining comments are fixed, I think this PR is ready to be merged. Thanks for working on this, @StevenXL.

**Comment by StevenXL** [comment]
@sebastinas sorry for the amount of time this has taken. I've updated the PR to reflect your suggestions. Let me know if there are other items that need work. I hope I merged the commits correctly.

**Pull request merged**